modules/router/templates/draft-assets.conf.erb

```diff
@@ -32,6 +32,10 @@ server {
   add_header "Access-Control-Allow-Methods" "GET, OPTIONS";
   add_header "Access-Control-Allow-Headers" "origin, authorization";
 
+  location /auth/gds/ {
+    proxy_pass <%= @upstream_ssl ? 'https' : 'http' %>://asset-manager.<%= @app_domain %>;
+  }
```

Comment:
I think this location regular expression might be too strict. Running `rake routes` in Asset Manager (see below), there's an `/auth/failure` route defined which we should probably support.

```
     gds_sign_in GET    /auth/gds/callback(.:format)              authentications#callback
    gds_sign_out GET    /auth/gds/sign_out(.:format)              authentications#sign_out
    auth_failure GET    /auth/failure(.:format)                   authentications#failure
                 PUT    /auth/gds/api/users/:uid(.:format)        api/user#update
                 POST   /auth/gds/api/users/:uid/reauth(.:format) api/user#reauth
```

Also looking at the output from the Asset Manager Rails app in my development environment (see below), it looks as if there is also an `/auth/gds` endpoint (see last line of the output) which we almost certainly need which wouldn't match the pattern you've specified. I've tried to work out what implements that endpoint, but have been unable to track it down.

```
19:03:05 asset-manager.1                                        | Started GET "/media/5a70bffe759b7464e73086ef/tmp.txt" for 127.0.0.1 at 2018-01-30 19:03:05 +0000
19:03:05 asset-manager.1                                        | Processing by MediaController#download as TEXT
19:03:05 asset-manager.1                                        |   Parameters: {"id"=>"5a70bffe759b7464e73086ef", "filename"=>"tmp.txt"}
19:03:05 asset-manager.1                                        | MONGODB | localhost:27017 | govuk_assets_development.find | STARTED | {"find"=>"assets", "filter"=>{"deleted_at"=>nil, "_id"=>BSON::ObjectId('5a70bffe759b7464e73086ef')}}
19:03:05 asset-manager.1                                        | MONGODB | localhost:27017 | govuk_assets_development.find | SUCCEEDED | 0.00060913s
19:03:05 asset-manager.1                                        | Redirected to http://draft-assets.dev.gov.uk/media/5a70bffe759b7464e73086ef/tmp.txt
19:03:05 asset-manager.1                                        | Completed 302 Found in 4ms
19:03:05 asset-manager.1                                        | 
19:03:05 asset-manager.1                                        | 
19:03:05 asset-manager.1                                        | 127.0.0.1 - - [30/Jan/2018:19:03:05 UTC] "GET /media/5a70bffe759b7464e73086ef/tmp.txt HTTP/1.0" 302 135
19:03:05 asset-manager.1                                        | - -> /media/5a70bffe759b7464e73086ef/tmp.txt
19:03:05 asset-manager.1                                        | Started GET "/media/5a70bffe759b7464e73086ef/tmp.txt" for ::1 at 2018-01-30 19:03:05 +0000
19:03:05 asset-manager.1                                        | Processing by MediaController#download as TEXT
19:03:05 asset-manager.1                                        |   Parameters: {"id"=>"5a70bffe759b7464e73086ef", "filename"=>"tmp.txt"}
19:03:05 asset-manager.1                                        | Authenticating with gds_sso strategy
19:03:05 asset-manager.1                                        | Completed   in 1ms
19:03:05 asset-manager.1                                        | 
19:03:05 asset-manager.1                                        | 
19:03:05 asset-manager.1                                        | ::1 - - [30/Jan/2018:19:03:05 UTC] "GET /media/5a70bffe759b7464e73086ef/tmp.txt HTTP/1.0" 302 105
19:03:05 asset-manager.1                                        | - -> /media/5a70bffe759b7464e73086ef/tmp.txt
19:03:05 asset-manager.1                                        | Started GET "/auth/gds" for 127.0.0.1 at 2018-01-30 19:03:05 +0000
19:03:05 asset-manager.1                                        | I, [2018-01-30T19:03:05.174122 #26138]  INFO -- omniauth: (gds) Request phase initiated.
19:03:05 asset-manager.1                                        | 127.0.0.1 - - [30/Jan/2018:19:03:05 UTC] "GET /auth/gds HTTP/1.0" 302 236
19:03:05 asset-manager.1                                        | - -> /auth/gds
```

Perhaps we should just support everything matching `/auth/`...? Or alternatively explicitly list each route.

Reply:
For completeness, I think I've tracked down where the `/auth/gds` route is handled. I'm pretty sure it's handled by the `Warden::Manager` Rack app which is added as middleware by `gds-sso`.